app/models/artefact.rb

```diff
@@ -401,8 +401,14 @@ def valid_url_path?(path)
 
   def validate_redirect_url
     return unless self.redirect_url.present?
-    unless valid_url_path?(self.redirect_url)
+    unless valid_redirect_url_path?(self.redirect_url)
       errors[:redirect_url] << "is not a valid redirect target"
     end
   end
+
+  def valid_redirect_url_path?(target)
+    target.starts_with?("/") && target !~ %r{//} && target !~ %r{./\z}
```

Comment:
This is missing the `URI.parse` step, and therefore will allow lots of invalid things.